cli/azd/pkg/environment/environment.go

```diff
@@ -160,14 +160,14 @@ func (e *Environment) DotenvSet(key string, value string) {
 	delete(e.deletedKeys, key)
 }
 
-// GetEnvName is shorthand for Getenv(EnvNameEnvVarName)
-func (e *Environment) GetEnvName() string {
-	return e.Getenv(EnvNameEnvVarName)
-}
+// Name gets the name of the environment
+// If empty will fallback to the value of the AZURE_ENV_NAME environment variable
+func (e *Environment) Name() string {
+	if e.name == "" {
```

Comment:
In what cases do we construct and return an environment with an empty name?  Is it possible for us to structure things such that `e.name` is always a non-empty string (ignoring the zero value of `Environment` which is already "invalid")?

Reply:
Basically this also gets us to a point where the retrieval of the environment name is always based on the value the provider specifies (folder names in local, etc)

The `AZURE_ENV_NAME` is then only really used to set environment name using `-e` flag, or used in string interpolation for bicep params or to make the environment available during hooks.

Reply:
Nothing changes - this change embraces that and basically just creates the azd environment variable if it is missing for any reason.